### ISSUE TITLE
Why? :)

### ISSUE THREAD

**Issue by odino** [huytd/dev.env#2]
Silly question: why this project? Syntax-wise it feels a lot like docker-compose, and having a dependency on docker / docker-compose shouldn't be painful :-)

**Comment by huytd** [comment]
nice question :smile: the answer is: Why not? 

I made this project for developers who lazy. he may lazy to install Docker or Vagrant, or even lazy to setup his own development environment.

I'm not against coding on docker/vagrant but it would be faster to just open terminal and type few lines of text to define your dev stack, then, voilà! everything is ready, just start coding, or run some snippets.

I also found myself, many times, forgot to start docker machine on my Mac and prefer to run some code just directly on my machine instead of a container or virtual machine.

**Comment by lackerman** [comment]
Why not use something established like ansible that already has a whole library of setup configs? Don't get me wrong, I like the idea of dev env consistency and ease of setup without any 3rd party help. Just thought it would be nice to dig a little deeper into the why?

**Comment by huytd** [comment]
Most of the tools nowaday seems to be focusing on devops, or at least, covering both development and deployment. Or heavy depend on devops solutions like Docker or Vagrant,... It would be a big concept at the beginning and some developer will try to avoid using them.

My goal is to creating a simple tool that everybody can use without trying to understand too much concept, somebody just need to have a quick environment to run some test snippets or quickly run multiple applications with different node or ruby version.

In another words, I want to make a pure-dev solution after spending quite a long time playing around with docker, vagrant...

TL;DR: Because Ansible or something has too many features, supporting too many things. I want a single tool that do only single thing that work! May be we can even "steal" some features from Ansible. :D

**Comment by odino** [comment]
:+1:

**Comment by hiendv** [comment]
For me, the only motivation to use containers or VMs is the separated environment.
It's not about the dependencies, it's about the separation.
Anyway, nice project.
